docs/endpoint.md

```diff
@@ -0,0 +1,106 @@
+# Endpoint
+
+Each request begins and ends its lifecycle inside your application in an endpoint. The endpoints handles starting the server and transforming request passing it through several defined plugs before calling the [Router](routing.html).
+
+Phoenix applications starts the HelloPhoenix.Web.Endpoint as a supervised process. By default, the Endpoint is added to the supervision tree in `lib/hello_phoenix/application.ex`
+
+```elixir
+defmodule HelloPhoenix.Application do
+  use Application
+  def start(_type, _args) do
+    #...
+
+    children = [
+      supervisor(HelloPhoenix.Web.Endpoint, []),
+    ]
+
+    opts = [strategy: :one_for_one, name: HelloPhoenix.Supervisor]
+    Supervisor.start_link(children, opts)
+  end
+end
+```
+
+### Endpoint Contents
+
+The endpoint includes plugs that are useful for all requests for our application.
```

Comment:
"The Endpoint's job is to hold plugs that are common to all requests coming into your application."